### PR TITLE
Add Static JSON Creator to DateTimeRfc1123

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Adds a package-private static method `JsonCreator` to `DateTimeRfc1123` to resolve deserialization failures in Azure Storage when self-closing or empty XML tags pass an empty string to `new DateTimeRfc1123(String)`. Jackson will now use the String-based static method when deserializing `DateTimeRfc1123`s which is able to handle the datetime string being null or empty to return the appropriate null value instead of a deserialization failure.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by ibrahimrabab — APPROVED** [review]

File: sdk/storage/azure-storage-blob/pom.xml [hunk 1]
```diff
@@ -76,7 +76,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
-      <version>1.27.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
+      <version>1.28.0-beta.1</version> <!-- {x-version-update;unreleased_com.azure:azure-core;dependency} -->
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/pom.xml:79]: this is going to block release? is there a workaround ?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/DebugTest.groovy [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.azure.core.util.serializer.JacksonAdapter
+import com.azure.core.util.serializer.SerializerEncoding
+import com.azure.storage.blob.models.BlobServiceStatistics
+import spock.lang.Specification
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.class DebugTest
+
+class DebugTest extends Specification {
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/DebugTest.groovy:11]: should this go to one of the core test suites ?

Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/DebugTest.groovy:11]: I'm planning on deleting this, didn't mean for it to be included in the commit.

File: sdk/storage/azure-storage-blob/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/pom.xml:79]: I'll see if there is anything that can be done without changes outside of Storage code

File: sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java [hunk 3]
```diff
@@ -47,6 +48,23 @@ public OffsetDateTime getDateTime() {
...
+     * If {@code date} is null or an empty string null will be returned.
+     *
+     * @param date RFC1123 datetime string.
+     * @return The DateTimeRfc1123 representation of the datetime string, or null if {@code date} is null or empty.
+     */
+    @JsonCreator
+    static DateTimeRfc1123 fromString(final String date) {
+        if (CoreUtils.isNullOrEmpty(date)) {
+            return null;
+        }
```
Comment (srnagar) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java:63]: This is different behavior from the constructor that takes a string input. I guess the constructor throws an exception and we should have the same behavior here.

Comment (alzimmermsft) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/DateTimeRfc1123.java:63]: That is correct that the behavior is different, in most cases empty string won't appear in the constructor given JSON handling is a bit better than XML (where `<XmlProperty></XmlProperty>` may be handled differently than `<XmlProperty/>`). Without this change the XmlMapper itself would need to have coercion configuration changes for DateTimeRfc1123 to treat empty string as null.

I think this change is more appropriate than having a coercion and matches the pattern we've seen more often with models in `azure-core` with JsonCreators.

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceAPITest.groovy [hunk 4]
```diff
@@ -763,7 +764,14 @@ class ServiceAPITest extends APISpec {
         response.getHeaders().getValue("x-ms-request-id") != null
         response.getHeaders().getValue("Date") != null
         response.getValue().getGeoReplication().getStatus() != null
-        response.getValue().getGeoReplication().getLastSyncTime() != null
+
+        // The LastSyncTime will return a DateTimeRfc1123 if the replication status is LIVE
+        // but there are two other statuses, unavailable and bootstrap, which will return null.
+        if (response.getValue().getGeoReplication().getStatus() == GeoReplicationStatus.LIVE) {
+            response.getValue().getGeoReplication().getLastSyncTime() != null
```
Comment (rickle-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceAPITest.groovy:771]: I think spock requires there be an "assert" at the beginning of the line if the check is inside a block like this

Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ServiceAPITest.groovy:771]: Otherwise it'll vacuously evaluate to true

**Review by srnagar — APPROVED** [review]

**Pull request merged**